### PR TITLE
refactor: cleanup dead code and some bug fixes

### PR DESCRIPTION
**Pull request by G3root**

### CODE REVIEW

File: src/app/signup/page.tsx [hunk 1]
```diff
@@ -1,19 +0,0 @@
-import { redirect } from "next/navigation";
-import { getServerSession } from "next-auth";
-import SignupForm from "@/components/onboarding/signup";
-import { authOptions } from "@/server/auth";
-import { type Metadata } from "next";
-
-export const metadata: Metadata = {
-  title: "OpenCap - Sign Up",
```
Comment (G3root) [inline src/app/signup/page.tsx:8]: we don't need a signup page now

File: src/providers/progress-bar.tsx [hunk 2]
```diff
@@ -6,7 +6,7 @@ type ProviderProps = {
   children: React.ReactNode;
 };
 
-const ProgressBarProvider = ({ children }: ProviderProps) => {
```
Comment (G3root) [inline src/providers/progress-bar.tsx:9]: refactored inside providers folder

File: next.config.js [hunk 3]
```diff
@@ -20,16 +20,6 @@ const config = {
 
     return config;
   },
-
-  async redirects() {
-    return [
-      {
-        source: "/",
```
Comment (G3root) [inline next.config.js:27]: remove double redirect. now the redirection is handle inside the home page

File: docker/docker-compose.yml [hunk 4]
```diff
@@ -32,7 +32,23 @@ services:
...
+  mc:
+    image: minio/mc
+    container_name: opencap-minio-client
+    entrypoint: >
+      /bin/sh -c "
+      /usr/bin/mc config;
+      /usr/bin/mc config host add myminio http://minio:9002 opencap password;
+      /usr/bin/mc mb myminio/opencap;
+      /usr/bin/mc mb myminio/opencap-public;
+      /usr/bin/mc anonymous set public myminio/opencap-public;
```
Comment (G3root) [inline docker/docker-compose.yml:47]: creates private bucket and public bucket

File: src/components/member/member-profile.tsx [hunk 5]
```diff
@@ -121,13 +120,9 @@ export const ProfileSettings = ({ memberProfile }: ProfileType) => {
         keyPrefix: "profile-avatars",
         identifier: session.user.id,
       };
-      const { key } = await uploadFile(file, options, "publicBucket");
+      const { bucketUrl } = await uploadFile(file, options, "publicBucket");
 
-      const s3BaseURL = `https://${env.PUBLIC_UPLOAD_BUCKET}.s3.amazonaws.com`;
```
Comment (G3root) [inline src/components/member/member-profile.tsx:126]: fixes hard coding of aws url and accessing env variables in client side error.

File: src/components/onboarding/company.tsx [hunk 6]
```diff
@@ -1,358 +0,0 @@
-"use client";
-
-import { Navbar } from "@/components/navbar";
-import { Button } from "@/components/ui/button";
-import { Input } from "@/components/ui/input";
-import { zodResolver } from "@hookform/resolvers/zod";
-import { RiArrowRightLine } from "@remixicon/react";
```
Comment (G3root) [inline src/components/onboarding/company.tsx:7]: dead code

**Review by dahal — APPROVED** [review]
💜

**Pull request merged**